kong/core/router.lua

```diff
@@ -464,7 +464,7 @@ function _M.new(apis)
     method = upper(method)
 
 
-    local host = headers["host"] or headers["Host"]
+    local host = headers["X-Host-Override"] or headers["x-host-override"] or headers["host"] or headers["Host"]
```

Comment:
> Since OpenResty lowercases the header names by default, we shouldn't need this extra condition?

This condition is for tests with some mock headers tables. By default, we lookup the lowercase name because it is the one provided by OR (Camel Case is a metatable lookup), but we have some unit tests providing a table with `{ ["Host"] = "example.com" }` for readability.